src/main/java/org/dataone/portal/TokenGenerator.java

```diff
@@ -181,35 +184,52 @@ public String getJWT(String userId, String fullName)
     private void setPrivateKey() throws IOException {
         String privateKeyFileName =
             Settings.getConfiguration().getString("cn.server.privatekey.filename");
-        String privateKeyPassword = null;
 
         CertificateManager cmInst = CertificateManager.getInstance();
         // consumers do not need the private key
         if (privateKeyFileName != null) {
-            privateKey = (RSAPrivateKey) cmInst.loadPrivateKeyFromFile(privateKeyFileName,
-                                                                       privateKeyPassword);
+            // uses a null password:
+            privateKey = (RSAPrivateKey) cmInst.loadPrivateKeyFromFile(privateKeyFileName, null);
         }
     }
 
     /*
      * Set the public key
      * @throws IOException
      */
-    private void setPublicKey() throws IOException {
-        // use either the configured certificate, or fetch it from the CN
-        String certificateFileName =
-            Settings.getConfiguration().getString("cn.server.publiccert.filename");
+    protected void setPublicKeys() throws IOException {
+
+        publicKeys = new ArrayList<>();
+        List<BigInteger> publicKeyModuli = new ArrayList<>();
+
+        // use the configured certificate, if it exists, and also the one fetched from the CN
+        String[] certificateFileNames =
+            Settings.getConfiguration().getStringArray("cn.server.publiccert.filename");
         CertificateManager cmInst = CertificateManager.getInstance();
-        log.debug("certificateFileName=" + certificateFileName);
-        if (certificateFileName != null && certificateFileName.length() > 0) {
-            publicKey =
-                (RSAPublicKey) cmInst.loadCertificateFromFile(certificateFileName).getPublicKey();
-        } else {
-            Certificate cert = fetchServerCertificate();
-            log.debug("using certificate from server: " + cert);
-            if (cert != null) {
-                publicKey = (RSAPublicKey) cert.getPublicKey();
-            }  // what happens if publicKey is null?
+        log.debug("certificateFileNames: \n" + Arrays.toString(certificateFileNames));
+        for (String certFileName : certificateFileNames) {
```

Comment:
This isn't necessary with the current implementation, since the call to `getStringArray("cn.server.publiccert.filename")` returns an empty array, not `null`, when:
* the key does not exist, and
* the key exists, and the value is set to `null`

I've added 2 more testcases for the above 2 scenarios.

However, I noticed in the api docs for apache commons, that the Configuration could return null if the key doesn't exist, which is confusing - so I'll also add a check here, to be safe, in case we upgrade in future and it breaks... :-) 

see commit 1c7b3339e7c072cddd2be61c7bdb07044c93f981